### ISSUE TITLE
How to fine-tune the model after layer-by-layer retraining?

### ISSUE THREAD

**Issue by PeiGiZhu** [joennlae/halutmatmul#19]
Really nice work! The idea of using H matrix to link soft encode and hard encode is impressive!

I notice the paper described three stages (pre-train, layer-by-layer, fine-tune) of training halutmatmul-based model. And the command example indicated that the train.py is for pre-training stage, and retraining.py is for layer-by-layer stage. I am wondering how to fine-tune the model after retraining it layer by layer? I see some comments in retraining.py related to fine-tuning, how can I use it properly for fine-tuning?

**Comment by joennlae** [comment]
Thank you for the nice words.

I agree it needs to be appropriately documented. I used the same script as for the layer-per-layer stage. It takes the checkpoint with all the replaced layers and then adjusts the learning rate + number of epochs according to the comments.

https://github.com/joennlae/halutmatmul/blob/4655152246e4a7203a9401bd6f4a71905e9cb682/src/python/retraining.py#L378-L380

Does that help you?

**Comment by PeiGiZhu** [comment]
Thanks, now I know how to do it. 

For others facing the same problem:

Fine-tuning uses the same script as the layer-by-layer, but EPOCH and LR are different.
The only difference in the retraining commands is using new pth generated in the layer-by-layer stage as a parameter for the -checkpoint. No need to worry about overwriting the original checkpoints.
